ssl/quic/quic_record_util.c

```diff
@@ -38,6 +39,15 @@ int ossl_quic_hkdf_extract(OSSL_LIB_CTX *libctx,
         || (kctx = EVP_KDF_CTX_new(kdf)) == NULL)
         goto err;
 
+    /*
+     * According to RFC 9000, the length of destination connection ID must be
+     * at least 8 bytes. It means that the length of destination connection ID
+     * may be less than the minimum length for HKDF required by FIPS provider.
```

Comment:
Not sure this should be done here.. It should probably break in FIPS mode.

Reply:
The operation may fail if FIPS provider is used with old libssl.so.

I think it cannot be avoided unless
- move the `ossl_quic_hkdf_extract` function into FIPS provider
- or has a global pedantic flag, which is turned off in default for back-compatibility, in fipsmodule.cnf

If I am wrong, please correct me. Thank you.

Reply:
@t8m or @paulidale  Do you have an opinion on this, technically it should fall over here..

Reply:
Yeah #$%@! FIPS rules bite again. The initial secret where this is used is not really security relevant so IMO OK to set pedantic to 0.